erpnext/controllers/selling_controller.py

```diff
@@ -559,6 +559,12 @@ def validate_target_warehouse(self):
 				frappe.throw(_("Row {0}: Delivery Warehouse ({1}) and Customer Warehouse ({2}) can not be same")
 					.format(d.idx, warehouse, warehouse))
 
+		if not self.get("is_internal_customer") and any(d.get("target_warehouse") for d in items):
+			msg = _("Target Warehouse set for some items but the customer is not an internal customer.")
+			msg += " " + _("This {} will be treated as material transfer.").format(_(self.doctype))
```

Comment:
Can this be a toast ? It will be disrupting for those who have such transfers often from UX POV.
```suggestion
			msg = _("Target Warehouse is set for some items but the customer is not an internal customer.")
			msg += " " + _("This {} will be treated as material transfer.").format(_(self.doctype))
```

Reply:
https://github.com/frappe/erpnext/commit/2cfafede4408c943d003a04985cadb869f811c21